appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -82,12 +82,37 @@ Blockly.WorkspaceSvg.prototype.onMouseDown_ = (function(func) {
     return func;
   } else {
     var f = function(e) {
-      var metrics = Blockly.common.getMainWorkspace().getMetrics();
-      var point = Blockly.utils.browserEvents.mouseToSvg(e, this.getParentSvg(), this.getInverseScreenCTM());
-      point.x = (point.x + metrics.viewLeft) / this.scale;
-      point.y = (point.y + metrics.viewTop) / this.scale;
-      this.latestClick = point;
-      return func.call(this, e);
+      try {
+        var metrics = Blockly.common.getMainWorkspace().getMetrics();
+        var point = Blockly.utils.browserEvents.mouseToSvg(e, this.getParentSvg(), this.getInverseScreenCTM());
+        point.x = (point.x + metrics.viewLeft) / this.scale;
+        point.y = (point.y + metrics.viewTop) / this.scale;
+        this.latestClick = point;
+        return func.call(this, e);
+      } finally {
+        //if drawer exists and supposed to close
+        if (this.drawer_ && this.drawer_.flyout_.autoClose) {
+          this.drawer_.hide();
+        }
+        if (this.backpack_ && this.backpack_.flyout_.autoClose) {
+          this.backpack_.hide();
+        }
+
+        //Closes mutators
+        var blocks = this.getAllBlocks();
+        var numBlocks = blocks.length;
+        var temp_block = null;
+        for(var i = 0; i < numBlocks; i++) {
+          temp_block = blocks[i];
+          if(temp_block.mutator){
+            //deselect block in mutator workspace
+            if(Blockly.common.getSelected() && Blockly.common.getSelected().workspace && Blockly.common.getSelected().workspace!=Blockly.common.getMainWorkspace()){
+              Blockly.common.getSelected().unselect();
+            }
+            blocks[i].mutator.setBubbleVisible(false);
+          }
+        }
+      }
```

Comment:
Note that this change was taken out intentionally in a fix for the Blockly update, so we should restore the previous logic here.